### ISSUE TITLE
Adding an icon crashes cx_Freeze on Windows

### ISSUE THREAD

**Issue by JaapJoris** [marcelotduarte/cx_Freeze#856]
I am trying to add an icon to my Windows executable. Here is the `setup.py` that I'm using:

```
from cx_Freeze import setup, Executable

# Write a simple test script to disk
with open('test.py', 'w') as f:
    f.write("input('hello world')")

setup(
    name='test',
    executables = [Executable(
        'test.py',
        targetName='test.exe',
        icon='test.ico',
    )],
)
```

When I run the command `python3 setup.py build` on Windows I get the following error:

```
running build
running build_exe
error during GetDependentFiles() of 'c:\\users\\super formosa\\appdata\\local\\microsoft\\windowsapps\\pythonsoftwarefoundation.python.3.9_qbz5n2kfra8p0\\python.exe': (0, 'The file cannot be accessed by the system', 'c:\\users\\super formosa\\appdata\\local\\microsoft\\windowsapps\\pythonsoftwarefoundation.python.3.9_qbz5n2kfra8p0\\python.exe', 1920, None)
copying c:\program files\windowsapps\pythonsoftwarefoundation.python.3.9_3.9.496.0_x64__qbz5n2kfra8p0\vcruntime140.dll -> build\exe.win-amd64-3.9\vcruntime140.dll
copying c:\program files\windowsapps\pythonsoftwarefoundation.python.3.9_3.9.496.0_x64__qbz5n2kfra8p0\python39.dll -> build\exe.win-amd64-3.9\python39.dll
copying c:\program files\windowsapps\pythonsoftwarefoundation.python.3.9_3.9.496.0_x64__qbz5n2kfra8p0\python3.dll -> build\exe.win-amd64-3.9\python3.dll
copying C:\Users\Super Formosa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\cx_Freeze\bases\Console.exe -> build\exe.win-amd64-3.9\test.exe
error: [WinError 2] The system cannot find the file specified: 'build\\exe.win-amd64-3.9\\test.exe'
```

However, if I remove the line `icon='test.ico'`, the build finishes successfully.

From the error message, it seems that cx_Freeze is trying to add the icon to an `.exe` file that does not exist yet. Or maybe something else is wrong. In any case, I hope I have provided enough informationto identify this bug. I will be happy to provide more information if needed.

**Comment by marcelotduarte** [comment]
I think that your icon is invalid, the wrong format. In issue #824, the icon is png with ico extension.
PR #851 avoid this situation and put a warning.

**Comment by JaapJoris** [comment]
You were right, the icon file (saved with Gimp) was invalid! I used https://icoconvert.com/ to convert it and now it works! The real issue here was the incorrect error message :)

**Comment by JaapJoris** [comment]
Sorry, I cheered too soon: the error message still persists:

    error: [WinError 2] The system cannot find the file specified: 'build\\exe.win-amd64-3.9\\test.exe'

Even with the icon converted by icoconvert.com. I mistakenly thought the problem was solved, but I still had the line `icon='test.ico',` commented out...

Could you provide me with a know valid icon file so I can test with that?

**Comment by marcelotduarte** [comment]
https://github.com/marcelotduarte/cx_Freeze/blob/master/cx_Freeze/samples/icon/icon.ico

**Comment by JaapJoris** [comment]
Thanks, I should have found that myself. Using that exact icon file and the following code (both in the same directory), I can reproduce the error every time. I am using Python 3.9.1 on Windows 10. Any ideas?

```
from cx_Freeze import setup, Executable

# Write a simple test script to disk
with open('test.py', 'w') as f:
    f.write("input('hello world')")

setup(
    name='test',
    executables = [Executable(
        'test.py',
        targetName='test.exe',
        icon='icon.ico',
    )],
)
```

**Comment by marcelotduarte** [comment]
For me it build, runs and show the icon in explorer.

**Comment by marcelotduarte** [comment]
The icon in a zip file:
[icon.zip](https://github.com/marcelotduarte/cx_Freeze/files/5751827/icon.zip)

**Comment by marcelotduarte** [comment]
The build dir:
[exe.win-amd64-3.9.zip](https://github.com/marcelotduarte/cx_Freeze/files/5751830/exe.win-amd64-3.9.zip)

**Comment by JaapJoris** [comment]
Strange. There must something peculiar about my setup. The final two lines of the error message are:

```
copying C:\Users\Super Formosa\AppData\Local\Packages\PythonSof
twareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packa
ges\Python39\site-packages\cx_Freeze\bases\Console.exe -> build
\exe.win-amd64-3.9\test.exe
error: [WinError 2] The system cannot find the file specified: 
'build\\exe.win-amd64-3.9\\test.exe'
```

I can confirm that the file `build\exe.win-amd64-3.9\test.exe` was created. When I run it, I get a `ModuleNotFoundError: No module named 'encodings'`, which is expected because the `lib/` subdirectory hasn't been created yet.

I also tried building the `samples/icon` project. First, I got an `TypeError: __init__() got an unexpected keyword argument 'target_name'`, but when I changed `target_name` to `targetName` in `samples/icon/setup.py`, I again got the `[WinError 2] The system cannot find the file specified: 'build\\exe.win-amd64-3.9\\test_1.exe'` error.

Really, I'm just guessing here, but why are there double backslashes in the path specified by the error message?

**Comment by JaapJoris** [comment]
I think I solved it! My version of Python was installed from the Windows store into `c:\\users\\super formosa\\appdata\\local\\microsoft\\windowsapps\\pythonsoftwarefoundation.python.3.9_qbz5n2kfra8p0\\python.exe` using their custom MSIX installer.

I have now uninstalled this application and downloaded the MSI installer from python.org, which is now installed in  `c:\users\super formosa\appdata\local\programs\python\python39\python.exe`. Using this new Python executable I can successfully build the `samples/icon/setup.py` example!

TL;DR: I reinstalled Python and now it works

**Comment by klensy** [comment]
NB: @marcelotduarte add note to docs about that: if python installed from windows store, there can be errors when freezing apps.

**Comment by marcelotduarte** [comment]
I'll test it, but the python docs say: https://docs.python.org/3/using/windows.html#known-issues

**Comment by marcelotduarte** [comment]
cx_Freeze 6.5 has just been released.

**Comment by timitos** [comment]
I can reproduce this problem in a msys2 environment with python 3.8.7 and cx_Freeze 6.5

**Comment by marcelotduarte** [comment]
@timitos I can't reproduce since 6.5. Can you show me more details? 
With the following sample:
https://github.com/marcelotduarte/cx_Freeze/blob/master/cx_Freeze/samples/icon
What are your results?

**Comment by timitos** [comment]
@marcelotduarte This is my result:

```
User@DESKTOP-N2BTLCA MINGW64 ~/icon
$ python setup.py build
running build
running build_exe
copying c:/msys64/mingw64/bin/libpython3.8.dll -> build/exe.mingw-3.8/libpython3.8.dll
copying c:/msys64/mingw64/bin/libwinpthread-1.dll -> build/exe.mingw-3.8/libwinpthread-1.dll
copying C:/msys64/mingw64/lib/python3.8/site-packages/cx_Freeze/bases/Console.exe -> build/exe.mingw-3.8/test_1.exe
error: [WinError 2] Das System kann die angegebene Datei nicht finden: 'build/exe.mingw-3.8/test_1.exe
```
It seems that cx_Freeze.util.AddIcon seems to fail somehow. I already checked the paths of the target and the icon and both are valid.

**Comment by marcelotduarte** [comment]
@timitos Are sure you have cx_Freeze 6.5? For me it shows a warning, that is correct for now.
Do you are using the mingw 64-bit, 32-bit or Msys prompt? I'm using 64-bit.
See my python -VV and build:

![remmina_teste6-vm-win10-rdp_192 168 6 56_202118-1604,344898](https://user-images.githubusercontent.com/12752334/104035953-9113fe80-51b1-11eb-9bc3-7e0d9178f5b8.png)

**Comment by timitos** [comment]
I am using 64bit prompt too. I am sure to have cx_Freeze 6.5. I updated everything in msys2 to the newest packages.
```
$ pacman -Q mingw-w64-x86_64-python-cx_Freeze
mingw-w64-x86_64-python-cx_Freeze 6.5-1
```

Maybe i am missing an package?
These are the packages i have installed:
```
$ pacman -Qqe
base
filesystem
mercurial
mingw-w64-x86_64-atk
mingw-w64-x86_64-gcc
mingw-w64-x86_64-gdk-pixbuf2
mingw-w64-x86_64-gobject-introspection
mingw-w64-x86_64-goocanvas
mingw-w64-x86_64-gtk3
mingw-w64-x86_64-librsvg
mingw-w64-x86_64-nsis
mingw-w64-x86_64-python-babel
mingw-w64-x86_64-python-cx_Freeze
mingw-w64-x86_64-python-dateutil
mingw-w64-x86_64-python-gobject
mingw-w64-x86_64-python-pip
mingw-w64-x86_64-python-setuptools
patch
python
tar
vim

**Comment by timitos** [comment]
I forgot the python version:
```
$ python -VV
Python 3.8.7 (default, Dec 27 2020, 14:55:43)  [GCC 10.2.0 64 bit (AMD64)]
```

**Comment by marcelotduarte** [comment]
@timitos cx_Freeze depends on importlib-metadata. It appears to me with `pip list`
I think the issue is another.
1. Try to uninstall and install cx_Freeze and build
2. Try to run it in a cmd session like me (open a msys session and the run cmd)

**Comment by marcelotduarte** [comment]
In a pure mingw64 session it runs for me....

**Comment by timitos** [comment]
@marcelotduarte I did like you said and uninstalled and reinstalled cx_Freeze and it works like a charm now :-)

```
$ pacman -Rns mingw-w64-x86_64-python-cx_Freeze
checking dependencies...

Packages (4) mingw-w64-x86_64-python-importlib-metadata-3.1.1-1  mingw-w64-x86_64-python-more-itertools-8.2.0-1  mingw-w64-x86_64-python-zipp-3.1.0-1
             mingw-w64-x86_64-python-cx_Freeze-6.5-1

Total Removed Size:  1.38 MiB

:: Do you want to remove these packages? [Y/n] Y
:: Processing package changes...
(1/4) removing mingw-w64-x86_64-python-cx_Freeze                                                         [#############################################################] 100%
(2/4) removing mingw-w64-x86_64-python-importlib-metadata                                                [#############################################################] 100%
(3/4) removing mingw-w64-x86_64-python-zipp                                                              [#############################################################] 100%
(4/4) removing mingw-w64-x86_64-python-more-itertools                                                    [#############################################################] 100%


$ pacman -S mingw-w64-x86_64-python3-cx_Freeze
resolving dependencies...
looking for conflicting packages...

Packages (4) mingw-w64-x86_64-python-importlib-metadata-3.1.1-1  mingw-w64-x86_64-python-more-itertools-8.2.0-1  mingw-w64-x86_64-python-zipp-3.1.0-1
             mingw-w64-x86_64-python-cx_Freeze-6.5-1

Total Installed Size:  1.38 MiB

:: Proceed with installation? [Y/n] Y
(4/4) checking keys in keyring                                                                           [#############################################################] 100%
(4/4) checking package integrity                                                                         [#############################################################] 100%
(4/4) loading package files                                                                              [#############################################################] 100%
(4/4) checking for file conflicts                                                                        [#############################################################] 100%
(4/4) checking available disk space                                                                      [#############################################################] 100%
:: Processing package changes...
(1/4) installing mingw-w64-x86_64-python-more-itertools                                                  [#############################################################] 100%
(2/4) installing mingw-w64-x86_64-python-zipp                                                            [#############################################################] 100%
(3/4) installing mingw-w64-x86_64-python-importlib-metadata                                              [#############################################################] 100%
(4/4) installing mingw-w64-x86_64-python-cx_Freeze                                                       [#############################################################] 100%
```
Thank you very much for your help!

It is working in a mingw64 session for me.
